xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java

```diff
@@ -841,47 +833,60 @@ static <X extends FeatureTask> void resolveSpace(final X task, final Callback<X>
   private static <X extends FeatureTask> Future<Space> resolveSpace(final X task) {
     try {
       //FIXME: Can be removed once the Space events are handled by the SpaceTaskHandler (refactoring pending ...)
-      if (task.space != null) //If the space is already given we don't need to retrieve it
+      if (task.space != null) { //If the space is already given we don't need to retrieve it
         return Future.succeededFuture(task.space);
+      }
 
       //Load the space definition.
       return Space.resolveSpace(task.getMarker(), task.getEvent().getSpace())
-          .compose(
-              space -> {
-                if (space != null) {
-                  if (space.getExtension() != null && task.getEvent() instanceof ContextAwareEvent && SUPER == ((ContextAwareEvent<?>) task.getEvent()).getContext())
-                    return switchToSuperSpace(task, space);
-                  task.space = space;
-                  //Inject the extension-map
-                  return space.resolveCompositeParams(task.getMarker()).compose(resolvedExtensions -> {
-                    Map<String, Object> storageParams = new HashMap<>();
-                    if (space.getStorage().getParams() != null)
-                      storageParams.putAll(space.getStorage().getParams());
-                    storageParams.putAll(resolvedExtensions);
-
-                    task.getEvent().setParams(storageParams);
-
-                    //Inject the minVersion from the space config
-                    if (task.getEvent() instanceof SelectiveEvent)
-                      ((SelectiveEvent<?>) task.getEvent()).setMinVersion(space.getMinVersion());
-
-                    //Inject the versionsToKeep from the space config
-                    if (task.getEvent() instanceof ContextAwareEvent)
-                      ((ContextAwareEvent<?>) task.getEvent()).setVersionsToKeep(space.getVersionsToKeep());
-
-                    return Future.succeededFuture(space);
-                  });
-                }
-                else
-                  return Future.succeededFuture();
-              },
-              t -> {
-                logger.warn(task.getMarker(), "Unable to load the space definition for space '{}' {}", task.getEvent().getSpace(), t);
-                return Future.failedFuture(new HttpException(INTERNAL_SERVER_ERROR, "Unable to load the resource definition", t));
-              }
-          );
-    }
-    catch (Exception e) {
+          .compose(space -> {
+            if (space == null) {
+              return Future.succeededFuture();
+            }
+
+            if (!(task instanceof FeatureTask.ModifySpaceQuery) && !space.isActive()) {
```

Comment:
This seems to be a duplicated check. Is this not done in the #checkSpaceIsActive() method already, right after resolving the space?

Reply:
checkSpaceIsActive has been removed and logic is now part of the space resolution